src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1201,8 +1203,9 @@ void Ekf2::run()
 			if (PX4_ISFINITE(_ev_odom.q[0])) {
 				ev_data.quat = matrix::Quatf(_ev_odom.q);
 
-				// orientation measurement error from parameters
-				if (PX4_ISFINITE(_ev_odom.pose_covariance[_ev_odom.COVARIANCE_MATRIX_ROLL_VARIANCE])) {
+				// orientation measurement error from ev_data or parameters
+				if (!_param_ekf2_ev_noise_en.get()
+				    && PX4_ISFINITE(_ev_odom.pose_covariance[_ev_odom.COVARIANCE_MATRIX_ROLL_VARIANCE])) {
```

Comment:
We should probably check that the yaw variance is finite here, not roll.

Reply:
Good point

Reply:
Done